src/vcpkg/configuration.cpp

```diff
@@ -535,6 +537,61 @@ namespace
 
 namespace vcpkg
 {
+    static ExpectedL<Optional<std::string>> get_baseline_from_git_repo(const VcpkgPaths& paths, StringView url)
+    {
+        auto res = paths.git_fetch_from_remote_registry(url, "HEAD");
+        if (auto p = res.get())
+        {
+            return Optional<std::string>(std::move(*p));
+        }
+        else
+        {
+            return msg::format(msgUpdateBaselineGitError, msg::url = url)
+                .appendnl()
+                .append_raw(Strings::trim(res.error()));
+        }
+    }
+
+    static constexpr StringLiteral BUILTIN_GIT_URL = "https://github.com/microsoft/vcpkg";
+
+    ExpectedL<Optional<std::string>> RegistryConfig::get_latest_baseline(const VcpkgPaths& paths) const
+    {
+        if (kind == RegistryConfigDeserializer::KIND_GIT)
+        {
+            return get_baseline_from_git_repo(paths, repo.value_or_exit(VCPKG_LINE_INFO));
+        }
+        else if (kind == RegistryConfigDeserializer::KIND_BUILTIN)
+        {
+            return get_baseline_from_git_repo(paths, BUILTIN_GIT_URL);
+        }
+        else
+        {
+            return baseline;
+        }
+    }
+
+    StringView RegistryConfig::registry_location() const
+    {
+        if (kind == RegistryConfigDeserializer::KIND_BUILTIN)
+        {
+            return BUILTIN_GIT_URL;
```

Comment:
Since this is only used for "pretty viewing" by the user, I think this can stay as the upstream URL.